src/Psa/Util.purs

```diff
@@ -17,6 +17,12 @@ padLeft width str = replicate (width - Str.length str) " " <> str
 padRight :: Int -> String -> String
 padRight width str = str <> replicate (width - Str.length str) " "
 
-iter_ :: forall m f a b. Foldable f => Applicative m => f a -> (Int -> a -> m b) -> m Unit
+iter_ :: forall m f a b. Foldable f => Monad m => f a -> (Int -> a -> m b) -> m Unit
 iter_ xs f = snd $ foldl go (Tuple 0 (pure unit)) xs where
-  go (Tuple i a) b = Tuple (i + 1) (a <* f i b)
+  go (Tuple i a) b = 
+    Tuple (i + 1) 
+      (do
+        _ <- a
+        _ <- f i b
+        pure unit
+        )
```

Comment:
This function isn't stack safe. It may work at 3000, but will eventually fail. You would need to replace this with `foldRecM` not `foldl`.

Reply:
Resolved